### ISSUE TITLE
Deliver does not support 13 inch ipad screen resolution: 2064x2752

### ISSUE THREAD

**Issue by peterdk** [fastlane/fastlane#22030]
### New Issue Checklist

- [x ] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I see the App Store now asks for 13 inch ipad screenshots. I used a iPad Pro M4 with resolution 2064x2752 (which is indicated in appstore connect as valid). However, I do get a error in Fastlane, that the resolution is not supported.
```
Error: ./fastlane/screenshots-lite/th/13 inch-04-FOO.png - Invalid screen size (Actual size is 2064x2752. See the specifications to fix https://help.apple.com/app-store-connect/#/devd274dd925)
```
The linked website states that this resolution is valid. So I guess it needs to be added to Fastlane.
##
### Command executed

```
upload_to_app_store(app_identifier: "bundle.foo, skip_binary_upload: true, skip_metadata: true, overwrite_screenshots:true, screenshots_path: './fastlane/screenshots-foo', ignore_language_directory_validation: true)


``` 
##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>
[00:23:40]: ------------------------------
[00:23:40]: --- Step: default_platform ---
[00:23:40]: ------------------------------
[00:23:40]: Driving the lane 'ios upload_screenshots_foo' 🚀
[00:23:40]: ---------------------------------
[00:23:40]: --- Step: upload_to_app_store ---
[00:23:40]: ---------------------------------
[00:23:40]: Successfully loaded 'fastlane/Deliverfile' 📄
[00:23:40]: No values defined in './fastlane/Deliverfile'
[00:23:40]: Login to App Store Connect (email@email)
[00:23:42]: Login successful

+--------------------------------------------------------------------+
|                      deliver 2.220.0 Summary                       |
+--------------------------------------+-----------------------------+
| app_identifier                       | bundle.id     |
| skip_binary_upload                   | true                        |
| skip_metadata                        | true                        |
| overwrite_screenshots                | true                        |
| screenshots_path                     | ./fastlane/screenshots-foo |
| ignore_language_directory_validation | true                        |
| metadata_path                        | ./fastlane/metadata         |
| username                             |          |
| platform                             | ios                         |
| edit_live                            | false                       |
| use_live_version                     | false                       |
| skip_screenshots                     | false                       |
| skip_app_version_update              | false                       |
| force                                | false                       |
| screenshot_processing_timeout        | 3600                        |
| sync_screenshots                     | false                       |
| submit_for_review                    | false                       |
| verify_only                          | false                       |
| reject_if_possible                   | false                       |
| version_check_wait_retry_limit       | 7                           |
| phased_release                       | false                       |
| reset_ratings                        | false                       |
| team_id                              |                      |
| dev_portal_team_id                   |                  |
| run_precheck_before_submit           | true                        |
| precheck_default_rule_level          | warn                        |
| precheck_include_in_app_purchases    | true                        |
+--------------------------------------+-----------------------------+

[00:23:43]: 🚫 Invalid screenshots were detected! Here are the reasons:
[00:23:43]: 🚫 Error: ./fastlane/screenshots-foo/zh-Hans/13 inch-01_foo.png - Invalid screen size (Actual size is 2064x2752. See the specifications to fix https://help.apple.com/app-store-connect/#/devd274dd925)
[00:23:43]: 🚫 Error: ./fastlane/screenshots-foo/zh-Hans/13 inch-02_foo.png - Invalid screen size (Actual size is 2064x2752. See the specifications to fix https://help.apple.com/app-store-connect/#/devd274dd925)
[..]
 </pre>
</details>

### Environment

<details>
  <pre> 
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 14.5                                        |
| Ruby                        | 2.7.8                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.39.3 (Apple Git-146)          |
| Installation Source         | ~/.rvm/gems/ruby-2.7.8/bin/fastlane         |
| Host                        | macOS 14.5 (23F79)                          |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.7.8/lib                |
| OpenSSL Version             | OpenSSL 1.1.1w  11 Sep 2023                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 15.4                                        |
| Swift Version               | 5.10                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | nl_NL.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:ios)

platform :ios do
  desc "Update metadata lite"
  lane :update_metadata_lite do 
    upload_to_app_store(metadata_path: "fastlane/metadata-lite",app_identifier:"nl.umito.ios.minipiano",  skip_binary_upload: true, skip_metadata: false, skip_screenshots:true)
  end
end

platform :ios do
  desc "Update metadata pro"
  lane :update_metadata_pro do
    upload_to_app_store(metadata_path: "fastlane/metadata-pro",app_identifier: "nl.umito.ios.minipiano.pro", skip_binary_upload: true, skip_metadata: false, skip_screenshots:true)
  end
end

platform :ios do
  desc "Generate new localized screenshots"
  lane :screenshots do
    snapshot(app_identifier: "nl.umito.ios.minipiano")
    # system("sips --rotate 90 screenshots/*/*.png")
#    upload_to_app_store(skip_binary_upload: true, skip_metadata: true)
  end
end

platform :ios do
  desc "Upload screenshots"
  lane :upload_screenshots_lite do
    
    upload_to_app_store(app_identifier: "nl.umito.ios.minipiano", skip_binary_upload: true, skip_metadata: true, overwrite_screenshots:true, screenshots_path: './fastlane/screenshots-lite', ignore_language_directory_validation: true)
  end
end



```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
apple_id("appledev@umito.nl") # Your Apple email address

itc_team_id("525444") # App Store Connect Team ID
team_id("V69H5Z8LUZ") # Developer Portal Team ID

# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.220.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| did_you_mean                    | 1.4.0        |
| executable-hooks                | 1.7.1        |
| bundler-unload                  | 1.0.2        |
| rubygems-bundler                | 1.4.5        |
| bundler                         | 2.3.13       |
| rake                            | 13.2.1       |
| base64                          | 0.2.0        |
| nkf                             | 0.2.0        |
| rexml                           | 3.2.6        |
| CFPropertyList                  | 3.0.7        |
| public_suffix                   | 5.0.5        |
| addressable                     | 2.8.6        |
| artifactory                     | 3.0.17       |
| atomos                          | 0.1.3        |
| aws-eventstream                 | 1.3.0        |
| aws-partitions                  | 1.929.0      |
| aws-sigv4                       | 1.8.0        |
| jmespath                        | 1.6.2        |
| aws-sdk-core                    | 3.196.1      |
| aws-sdk-kms                     | 1.81.0       |
| aws-sdk-s3                      | 1.151.0      |
| babosa                          | 1.0.4        |
| claide                          | 1.1.0        |
| colored                         | 1.2          |
| colored2                        | 3.1.2        |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| declarative                     | 0.0.20       |
| digest-crc                      | 0.6.5        |
| domain_name                     | 0.6.20240107 |
| dotenv                          | 2.8.1        |
| emoji_regex                     | 3.2.3        |
| excon                           | 0.110.0      |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-httpclient              | 1.0.1        |
| multipart-post                  | 2.4.1        |
| faraday-multipart               | 1.0.4        |
| faraday-net_http                | 1.0.1        |
| faraday-net_http_persistent     | 1.2.0        |
| faraday-patron                  | 1.0.0        |
| faraday-rack                    | 1.0.0        |
| faraday-retry                   | 1.0.3        |
| ruby2_keywords                  | 0.0.5        |
| faraday                         | 1.10.3       |
| http-cookie                     | 1.0.5        |
| faraday-cookie_jar              | 0.0.7        |
| faraday_middleware              | 1.2.0        |
| fastimage                       | 2.3.1        |
| gh_inspector                    | 1.1.3        |
| jwt                             | 2.8.1        |
| multi_json                      | 1.15.0       |
| os                              | 1.1.4        |
| signet                          | 0.19.0       |
| googleauth                      | 1.8.1        |
| httpclient                      | 2.8.3        |
| mini_mime                       | 1.1.5        |
| trailblazer-option              | 0.1.2        |
| uber                            | 0.1.0        |
| representable                   | 3.2.0        |
| retriable                       | 3.1.2        |
| google-apis-core                | 0.11.3       |
| google-apis-androidpublisher_v3 | 0.54.0       |
| google-apis-playcustomapp_v1    | 0.13.0       |
| google-cloud-env                | 1.6.0        |
| google-apis-iamcredentials_v1   | 0.17.0       |
| google-apis-storage_v1          | 0.31.0       |
| google-cloud-errors             | 1.4.0        |
| google-cloud-core               | 1.7.0        |
| google-cloud-storage            | 1.47.0       |
| json                            | 2.7.2        |
| mini_magick                     | 4.12.0       |
| naturally                       | 2.2.1        |
| optparse                        | 0.5.0        |
| plist                           | 3.7.1        |
| rubyzip                         | 2.3.2        |
| security                        | 0.1.5        |
| simctl                          | 1.6.10       |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 2.5.0        |
| terminal-table                  | 3.0.2        |
| tty-screen                      | 0.8.2        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| word_wrap                       | 1.0.0        |
| nanaimo                         | 0.3.0        |
| xcodeproj                       | 1.24.0       |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| xcpretty-travis-formatter       | 1.0.1        |
</details>

*generated on:* **2024-05-16**
</details></pre>
</details>

**Comment by peterdk** [comment]
I did got to work around it a bit by adding iPad Pro (12.9-inch) (3rd generation) screenshots, that seem also to fall under that category of 13 inch ones. But still the supported resolutions need a update I guess.

**Comment by hekuli** [comment]
same here

**Comment by gongzhang** [comment]
Same here. Unable to upload 12.9-inch screenshots. The 12.9-inch screenshots are uploaded to the latest 13-inch iPad incorrectly.

**Comment by gongzhang** [comment]
A quick dirty fix: https://github.com/fastlane/fastlane/compare/master...gongzhang:fastlane:master

The most confusing part is that every `IOS_IPAD_PRO_12_9` in the code actually refers to the latest 13-inch iPad Pro, while every `IOS_IPAD_PRO` in the code actually refers to the older 12.9-inch iPad Pro.

Please let me know if I was wrong.

**Comment by Charlinjoeaht** [comment]
When will this bug be availble in official fastlane release?

**Comment by ricsantos** [comment]
Just hit this, so bumping for visibility / priority.

`iPad Pro 13-inch (M4)-screenshot-5.png - Invalid screen size (Actual size is 2064x2752)`

**Comment by Mykyta-Chernenko** [comment]
it doesn't seem to be possible still

[20:42:11]: 🚫 Error: ./fastlane/screenshots/zh-Hant/iPad13_screenshot8.png - Invalid screen size (Actual size is 2064x2752. See the specifications to fix https://help.apple.com/app-store-connect/#/devd274dd925)

**Comment by bkrijgsman** [comment]
Same here!

**Comment by msangals** [comment]
+1

**Comment by bkrijgsman** [comment]
I btw fixed this issue by renaming the 13-inch screenshot and putting APP_IPAD_PRO_3GEN_129 in the filename.

Format:
2732x2048

Then it showed up under the 13 ich in Appstore Connect!

**Comment by BobDG** [comment]
@bkrijgsman this solves it! Crazy how simply renaming the screenshot works, but thanks!

**Comment by J7779** [comment]
Also bumping, @fastlaneTeam this is an extremely easy fix, please add the already merged PR into a general release.

**Comment by netbe** [comment]
@bkrijgsman this solved my issue too

**Comment by prkl78** [comment]
> I btw fixed this issue by renaming the 13-inch screenshot and putting APP_IPAD_PRO_3GEN_129 in the filename.
> 
> Format: 2732x2048
> 
> Then it showed up under the 13 ich in Appstore Connect!

I tried that, but still getting error :( 

`[00:33:44]: 🚫 Error: ./screenshots/zh-Hans/APP_IPAD_PRO_3GEN_129-1.jpg - Invalid screen size (Actual size is 2064x2752. See the specifications to fix https://help.apple.com/app-store-connect/#/devd274dd925)`

**Comment by FredericRuaudel** [comment]
@bkrijgsman The renaming trick doesn't seems to work for me… What did I do wrong?

`Error: ./fastlane/screenshots/de-DE/APP_IPAD_PRO_3GEN_129-01MainScreenLight_framed.png - Invalid screen size (Actual size is 2752x2064. See the specifications to fix https://help.apple.com/app-store-connect/#/devd274dd925)`

🤔

**Comment by bas-krijgsman-spotler** [comment]
@FredericRuaudel did you put the format on 2732x2048?

**Comment by FredericRuaudel** [comment]
> [@FredericRuaudel](https://github.com/FredericRuaudel) did you put the format on 2732x2048?

No, I thought the trick was just to rename it so it is downloaded with the correct new format… whatever, I'll do it manually until this fix is published. Thanks for your quick answer!

**Comment by danieljbk** [comment]
How has this not been fixed for 1.5 years?

**Comment by assembleMHN** [comment]
Is fastlane dying ? Its kinda strange that the good people behind it would let a bug like this where in 1.5 years you cannot support image upload of new required screen sizes ?

**Comment by stodirascu** [comment]
I'd mark this as closed as to make it work is simple as @bkrijgsman pointed out [here](https://github.com/fastlane/fastlane/issues/22030#issuecomment-2747064630)

The 2 iPad formats have some common dimensions and the name is used to disambiguate.

**Comment by assembleMHN** [comment]
> I'd mark this as closed as to make it work is simple as [@bkrijgsman](https://github.com/bkrijgsman) pointed out [here](https://github.com/fastlane/fastlane/issues/22030#issuecomment-2747064630)
> 
> The 2 iPad formats have some common dimensions and the name is used to disambiguate.

No the issue is not that it cannot be done, the issue is that Fastlane does not support Apples official screenshot sizes, you cannot upload dimentions "2064x2752" with fastlane, and this is a official support screenshot size for iPad in App Store Connect!

Below is the screen sizes that Apple supports for app store connect, these should all be supported by fastlane, if you try to upload 2064x2752 with fastlane it will tell you that its a unsupported format, which is not true. Its a bug, this has not been solved!


https://developer.apple.com/help/app-store-connect/reference/app-information/screenshot-specifications/


So the the main issue of this ticket has not been solved its still not possible to upload screenshots with the size of "2064x2752"

**Comment by sirghi** [comment]
You're right indeed @assembleMHN. On this occasion I've done some general cleanup in this area, and included the missing screenshots here #29760.
